File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.java
class Solution {
    public int solution(int[][] arr) {
        int answer = 0;
        int len = arr.length;
        
        for(int i = 0; i < len; i++){
            for(int j = 0; j < len; j++){
                if(arr[i][j] != arr[j][i]) return answer;
            }
        }
        
        return answer = 1;
    }
}

File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int** arr, size_t arr_rows, size_t arr_cols) {
    int answer = 0;
    
    for(int i = 0; i < arr_rows; i++){
        for(int j = 0; j < arr_cols; j++){
            if(arr[i][j] != arr[j][i]) {
                return answer;
            }
        }
    }
    
    answer++;
    
    return answer;
}